cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -490,6 +504,43 @@ private void closeScanner() {
     }
   }
 
+  /**
+   * Filters elements matching an index  {@link ScannerIterator}.
+   */
+  @VisibleForTesting
+  static final class FilterByIndexIterator extends AbstractCloseableIterator<StructuredRow> {
+    private final LimitIterator limitIterator;
```

Comment:
The Limiting iterator should be wrapping the filter iterator and not the other way around. Otherwise it is possible to return fewer elements than it should.

Reply:
Agree, the `limit` should have effect on the filtered result

Reply:
Yes, agreed. Will address